scripts/evaluation/auto_evaluator.py

```diff
@@ -0,0 +1,213 @@
+                                step=step,
+                                downstream_dir_path=self.downstream_paths["medQA"],
+                                checkpoint_path=checkpoint_path,
+                            ).evaluate()
+                            NEREBMPICOTaskEvaluator(
+                                evaluator_state=self.state,
+                                run=run,
+                                artifact=artifact,
+                                step=step,
+                                downstream_dir_path=self.downstream_paths["nerBC5CDR"],
+                                checkpoint_path=checkpoint_path,
+                            ).evaluate()
+                            NERBC5CDRTaskEvaluator(
+                                evaluator_state=self.state,
+                                run=run,
+                                artifact=artifact,
+                                step=step,
+                                downstream_dir_path=self.downstream_paths["nerEBMPICO"],
+                                checkpoint_path=checkpoint_path,
+                            ).evaluate()
+
+                    # TODO remove this later -Tony
+                    pdb.set_trace()
+
+    def prepare(
+        self, run: Run, artifact: Artifact, hyperparameter_artifact: Artifact
+    ) -> str:
+        """
+        Prepares artifact by following these steps:
+        1. Downloads the artifact from wandb.
+        2. Downloads the hyperparameter artifact from wandb (necessary for step 4)
+        3. Extracts the content from the tar file
+        4. Converts the DeepSpeed checkpoint to a HF checkpoint
+        5. Marks that this specific artifact has been already prepped.
+        """
+
+        # Create the sub directories and download the hyperparameters used for the particular run
+        run_dir_path: str = clean_path(os.path.join(self.output_dir, run.name))
+        checkpoint_dir_path: str = clean_path(os.path.join(run_dir_path, artifact.name))
+
+        # If we already prepared the artifact, then do nothing
+        if self.state.has_prepared(run.name, artifact.name):
+            hlog("Already downloaded artifact and converted to HuggingFace model.")
+            return checkpoint_dir_path
+
+        os.makedirs(checkpoint_dir_path, exist_ok=True)
+        hyperparameter_artifact.download(root=run_dir_path)
+
+        # Download the artifact from wandb
+        artifact.download(root=checkpoint_dir_path)
+
+        files: List[str] = os.listdir(checkpoint_dir_path)
+        if len(files) == 0:
+            hlog("Nothing found in the directory.")
+            return checkpoint_dir_path
+
+        tar_file_path: str = os.path.join(checkpoint_dir_path, files[0])
+        extract_tar_file(tar_file_path, checkpoint_dir_path)
+
+        deepspeed_checkpoint_path: str = os.path.join(checkpoint_dir_path, "deepspeed")
+        files: List[str] = os.listdir(deepspeed_checkpoint_path)
+        if len(files) == 0:
```

Comment:
if there are no files (or if the conversion fails), then the process is just goign to keep checking right? should we mark prepared here (and add a failed state for failures?)